include/bluetooth/audio/has.h

```diff
@@ -0,0 +1,350 @@
+/*
+ * Copyright (c) 2022 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_BLUETOOTH_AUDIO_HAS_H_
+#define ZEPHYR_INCLUDE_BLUETOOTH_AUDIO_HAS_H_
+
+/**
+ * @brief Hearing Access Service (HAS)
+ *
+ * @defgroup bt_has Hearing Access Service (HAS)
+ *
+ * @ingroup bluetooth
+ * @{
+ *
+ * The Hearing Access Service is used to identify a hearing aid and optionally
+ * to control hearing aid presets. This API implements the server functionality.
+ *
+ * [Experimental] Users should note that the APIs can change as a part of
+ * ongoing development.
+ */
+
+#include <stdint.h>
+#include <sys/util.h>
+#include <bluetooth/conn.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if defined(CONFIG_BT_HAS)
+#define BT_HAS_PRESET_CNT CONFIG_BT_HAS_PRESET_CNT
+#else
+#define BT_HAS_PRESET_CNT 0
+#endif /* CONFIG_BT_HAS */
+
+#define BT_HAS_PRESET_NAME_MIN 1
+#define BT_HAS_PRESET_NAME_MAX 40
+
+/** @brief Opaque Hearing Access Service object. */
+struct bt_has;
+
+/** Preset Properties values */
+enum bt_has_prop {
+	/** No properties set */
+	BT_HAS_PROP_NONE        = 0,
+
+	/** Preset name can be written by the client */
+	BT_HAS_PROP_WRITABLE    = BIT(0),
+
+	/** Preset availability */
+	BT_HAS_PROP_AVAILABLE   = BIT(1),
+};
+
+/** @brief Register structure for preset. */
+struct bt_has_preset_register_param {
+	/** Preset index */
+	uint8_t index;
+	/** Preset properties */
+	enum bt_has_prop properties;
+	/** Preset name */
+	const char *name;
+};
+
+/** @brief Preset operations structure. */
+struct bt_has_ops {
+	/** @brief Preset set active callback
+	 *
+	 *  Called if the client requests to activate preset identified by index.
+	 *
+	 *  @param has Pointer to the Hearing Access Service object.
+	 *  @param index Preset index requested to activate,
+	 *  @param sync Whether the server must relay this change to the
+	 *             other member of the Binaural Hearing Aid Set.
+	 *
+	 *  @return 0 in case of success or negative value in case of error.
+	 *  @return -EBUSY if operation cannot be executed at the time.
+	 *  @return -EINPROGRESS in case where user has to confirm once the requested preset
+	 *                       becomes active by calling @ref bt_has_preset_active_set.
```

Comment:
Not fully clear to me.  Does the preset become active by calling `bt_has_preset_active_set`?  Or does the user confirm by calling `bt_has_preset_active_set`?
How does the user confirm?  And does that happen before or after the call to `bt_has_preset_active_set`?